### PR TITLE
Add date to logging messages (#1623)

### PR DESCRIPTION
**Pull request by zeng-zc**
## Motivation

Please refer to https://github.com/sgl-project/sglang/issues/1623

## Modifications

* add date to engine logging
* add more info to uvicorn logging

## Checklist

- [x] Format your code according to the [Contributor Guide](https://github.com/sgl-project/sglang/blob/main/docs/en/contributor_guide.md).
- [ ] Add unit tests as outlined in the [Contributor Guide](https://github.com/sgl-project/sglang/blob/main/docs/en/contributor_guide.md).
- [ ] Update documentation as needed, including docstrings or example tutorials.

### CODE REVIEW

File: python/sglang/srt/server.py [hunk 1]
```diff
@@ -428,7 +428,11 @@ def launch_server(
     t.start()
 
     try:
+        from uvicorn.config import LOGGING_CONFIG
         # Listen for HTTP requests
+        LOGGING_CONFIG["formatters"]["default"]["fmt"] = "%(asctime)s [%(name)s] %(levelprefix)s %(message)s"
```
Comment (ByronHsu) [inline python/sglang/srt/server.py]: does this show the date as well?

Comment (zeng-zc) [reply python/sglang/srt/server.py]: Yes.
https://github.com/sgl-project/sglang/issues/1623#issuecomment-2413166385
In the above link, there is a comparison bewteen old and new log messages.

Comment (ByronHsu) [reply python/sglang/srt/server.py]: is it possible to include the timezone info?

File: python/sglang/srt/server.py [hunk 2]
```diff
@@ -428,7 +428,11 @@ def launch_server(
     t.start()
 
     try:
+        from uvicorn.config import LOGGING_CONFIG
```
Comment (merrymercy) [inline python/sglang/srt/server.py]: move import to the top of this file.

File: python/sglang/srt/server.py [hunk 3]
```diff
@@ -429,6 +429,12 @@ def launch_server(
 
     try:
         # Listen for HTTP requests
+        LOGGING_CONFIG["formatters"]["default"][
+            "fmt"
+        ] = "%(asctime)s [%(name)s] %(levelprefix)s %(message)s"
+        LOGGING_CONFIG["formatters"]["access"][
+            "fmt"
+        ] = '%(asctime)s [%(name)s] %(levelprefix)s %(client_addr)s - "%(request_line)s" %(status_code)s'
```
Comment (merrymercy) [inline python/sglang/srt/server.py]:
```suggestion
        ] = "[%(asctime)s] %(levelprefix)s %(message)s"
        LOGGING_CONFIG["formatters"]["access"][
            "fmt"
        ] = '[%(asctime)s] %(levelprefix)s %(client_addr)s - "%(request_line)s" %(status_code)s'
```

**Pull request merged**